<!DOCTYPE html>
<html lang="en"><head>
<script src="slides_files/libs/clipboard/clipboard.min.js"></script>
<script src="slides_files/libs/quarto-html/tabby.min.js"></script>
<script src="slides_files/libs/quarto-html/popper.min.js"></script>
<script src="slides_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="slides_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="slides_files/libs/quarto-html/light-border.css" rel="stylesheet">
<link href="slides_files/libs/quarto-html/quarto-syntax-highlighting-07ba0ad10f5680c660e360ac31d2f3b6.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="slides_files/libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="slides_files/libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="slides_files/libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.6.33">

  <meta name="author" content="Aurélien Beaude">
  <meta name="dcterms.date" content="2024-06-12">
  <title>Developping and exploring the interest of deep learning approaches in the field of multi-omics data</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="slides_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="slides_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
  </style>
  <link rel="stylesheet" href="slides_files/libs/revealjs/dist/theme/quarto-097ee88916ff14e1325742309d2793f9.css">
  <link href="slides_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="slides_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="slides_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="slides_files/libs/revealjs/plugin/reveal-pointer/pointer.css" rel="stylesheet">
  <link href="slides_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
  <script type="text/x-mathjax-config">
    MathJax.Hub.Config({ TeX: { extensions: ["color.js"] }});
  </script>
  <script type="text/x-mathjax-config">
      MathJax.Hub.Register.StartupHook("TeX color Ready", function() {
           MathJax.Extension["TeX/color"].colors["query"] = '#3399FF';
           MathJax.Extension["TeX/color"].colors["key"] = '#FF3333';
           MathJax.Extension["TeX/color"].colors["value"] = '#97D077';
      });
  </script>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">Developping and exploring the interest of deep learning approaches in the field of multi-omics data</h1>
  <p class="subtitle">Développer et explorer l’intérêt des approches de deep learning dans le domaine des données multi-omiques</p>

  <p class="date">2024-06-12</p>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Aurélien Beaude 
</div>
</div>
</div>
</section>
<section id="towards-a-personalized-medicine" class="slide level2">
<h2>Towards a personalized medicine</h2>
<div data-animate="" data-src="IntroductionPersoMedicine-MédecineTrad_Perso.svg"><!-- setup:
  - element: "#Title1"
    modifier: attr
    parameters:
      - class: fragment
        data-fragment-index: "0"
  - element: "#Patient1"
    modifier: attr
    parameters:
      - class: fragment
        data-fragment-index: "1"
  - element: "#Diagnosis1"
    modifier: attr
    parameters:
      - class: fragment
        data-fragment-index: "2"
  - element: "#Prognosis1"
    modifier: attr
    parameters:
      - class: fragment
        data-fragment-index: "3"
  - element: "#Treatment1"
    modifier: attr
    parameters:
      - class: fragment
        data-fragment-index: "3"
  - element: "#TreatmentOK"
    modifier: attr
    parameters:
      - class: fragment
        data-fragment-index: "4"
  - element: "#TreatmentKO"
    modifier: attr
    parameters:
      - class: fragment
        data-fragment-index: "5"
  - element: "#Title2"
    modifier: attr
    parameters:
      - class: fragment
        data-fragment-index: "6"
  - element: "#Patient2"
    modifier: attr
    parameters:
      - class: fragment
        data-fragment-index: "6"
  - element: "#Diagnosis2"
    modifier: attr
    parameters:
      - class: fragment
        data-fragment-index: "7"
  - element: "#Prognosis2"
    modifier: attr
    parameters:
      - class: fragment
        data-fragment-index: "8"
  - element: "#Treatment2"
    modifier: attr
    parameters:
      - class: fragment
        data-fragment-index: "9" --></div>
</section>
<section id="omics-data" class="slide level2">
<h2>Omics data</h2>

<img data-src="Omics.svg" class="quarto-figure quarto-figure-center r-stretch"></section>
<section id="precision-medicine" class="slide level2">
<h2>Precision medicine</h2>

<img data-src="IntroductionPersoMedicine-SlideIntro.svg" class="r-stretch quarto-figure quarto-figure-center" style="width:110.0%"></section>
<section id="outline" class="slide level2">
<h2>Outline</h2>
<div>

</div>
<div class="quarto-layout-panel" data-layout-ncol="3">
<div class="quarto-layout-row">
<div id="first-column" class="quarto-layout-cell" style="flex-basis: 33.3%;justify-content: flex-start;">
<h3 id="single-omics">Single omics</h3>
<div style="font-size: 70%; padding-left: 0.7em;">
<ul>
<li>State of the Art</li>
<li>AttOmics</li>
<li>Results</li>
</ul>
</div>
</div>
<div id="second-column" class="quarto-layout-cell" style="flex-basis: 33.3%;justify-content: flex-start;">
<h3 id="multi-omics">Multi Omics</h3>
<div style="font-size: 70%; padding-left: 0.7em;">
<ul>
<li>State of the Art</li>
<li>CrossAttOmics</li>
<li>Results</li>
<li>CrossAttOmicsGate</li>
</ul>
</div>
</div>
<div id="third-column" class="quarto-layout-cell" style="flex-basis: 33.3%;justify-content: flex-start;">
<h3 id="interpretability">Interpretability</h3>
<div style="font-size: 70%; padding-left: 0.7em;">
<ul>
<li>Counterfactuals</li>
<li>GAN</li>
<li>Results</li>
</ul>
</div>
</div>
</div>
</div>
<h3 id="conclusions-and-perspectives">Conclusions and Perspectives</h3>
</section>
<section>
<section id="single-omics-1" class="title-slide slide level1 center">
<h1>Single omics</h1>

</section>
<section id="state-of-the-art" class="slide level2">
<h2>State of the art</h2>

<img data-src="SOTA_single_omics.svg" class="quarto-figure quarto-figure-center r-stretch"></section>
<section id="limits" class="slide level2 smaller">
<h2>Limits</h2>
<h3 id="classical-deep-learning-limits">Classical deep learning limits</h3>
<div style="font-size: 90%;">
<ul>
<li>High number of parameters</li>
<li>With limited number of examples</li>
<li>Risk of overfitting</li>
</ul>
</div>
<div class="callout callout-important callout-style-simple">
<div class="callout-body">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-content">
<p><strong>How to reduce the number of parameters ?</strong></p>
</div>
</div>
</div>
<h3 id="gene-expression-impacts-patient-differently">Gene expression impacts patient differently</h3>
<div style="font-size: 90%;">
<ul>
<li>With classical DL, features interactions are learned during training and fixed for inference</li>
</ul>
</div>
<div class="callout callout-important callout-style-simple">
<div class="callout-body">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-content">
<p><strong>How to compute interaction specific to each patient ?</strong></p>
</div>
</div>
</div>
</section>
<section id="self-attention-mechanism" class="slide level2">
<h2>(Self)-Attention mechanism</h2>
<div class="columns">
<div class="column" style="width:75%;">
<div class="callout callout-note callout-style-simple">
<div class="callout-body">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-content">
<p>How to pick relevant information from input data <span class="math inline">\(X = [x_i]_{1 \leq i \leq L}\)</span> ?</p>
</div>
</div>
</div>
<div style="font-size: 65%;">
<ul>
<li><strong>Self-attention</strong> allows to capture the associated context of each input element by interacting with other elements <span class="math display">\[
\operatorname{Attention}\left({\color{query}Q},{\color{key}K},{\color{value}V}\right) = \operatorname{softmax}\left(\frac{{\color{query}Q}{\color{key}K^T}}{\sqrt{d_k}}\right){\color{value}V}
\]</span> <span class="math inline">\(\color{query}Q = \left[q_i\right]_{1 \leq i \leq L}\)</span> with <span class="math inline">\(\color{query}q_i = X_i \cdot W_q\)</span>, <span class="math inline">\(\color{key}K\)</span> and <span class="math inline">\(\color{value}V\)</span> are obtained similarly.</li>
<li>Quadratic complexity (Space and Time)</li>
</ul>
</div>
<div style="font-size: 70%">
<div class="callout callout-important callout-style-simple">
<div class="callout-body">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-content">
<p><strong>How to apply self-attention to large large vectors ?</strong></p>
</div>
</div>
</div>
<div class="callout callout-tip callout-style-simple">
<div class="callout-body">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-content">
<p>Group related features together and apply self-attention on it</p>
</div>
</div>
</div>
</div>
</div><div class="column" style="width:25%;">
<div class="quarto-figure quarto-figure-right">
<figure>
<p><img data-src="Attention.svg" class="quarto-figure quarto-figure-right" style="width:90.0%"></p>
</figure>
</div>
</div></div>
</section>
<section id="attomics-architecture" class="slide level2">
<h2>AttOmics Architecture</h2>
<div class="columns">
<div class="column" style="width:60%;">
<div class="quarto-figure quarto-figure-left">
<figure>
<p><img data-src="AttOmics-ArchitectureFull.svg" class="quarto-figure quarto-figure-left" style="width:90.0%"></p>
</figure>
</div>
</div><div class="column" style="width:40%;">
<div class="r-stack">
<div class="fragment fade-in-then-out" data-fragment-index="1">
<div style="font-size: 70%;">
<p><strong>Grouping Strategies</strong></p>
<ul>
<li>Random</li>
<li>Clustering</li>
<li>Knowledge:
<ul>
<li>Gene ontology</li>
<li>Cancer Hallmarks</li>
</ul></li>
</ul>
<p><span class="math display">\[
\begin{align}
X_G &amp;= \mathcal{T}\left(X\right) \\
&amp;= \left[X_{g_1}, \cdots, X_{g_4}\right]
\end{align}
\]</span></p>
</div>
</div>
<div class="fragment fade-out" data-fragment-index="5">
<div style="font-size: 70%;">
<div class="fragment fade-in" data-fragment-index="2">
<p><strong>Intra-group interactions</strong></p>
</div>
<div class="fragment fade-in" data-fragment-index="3">
<div class="callout callout-important callout-style-simple">
<div class="callout-body">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-content">
<p><strong>Gene grouping creates unwanted restrictions</strong></p>
<p>Genes in group <span class="math inline">\(g_1\)</span> cannot interact with genes from other groups</p>
</div>
</div>
</div>
</div>
<div class="fragment fade-in" data-fragment-index="4">
<div class="callout callout-tip callout-style-simple">
<div class="callout-body">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-content">
<p>Restore group interactions with the Attention mechanism</p>
</div>
</div>
</div>
</div>
<div class="fragment fade-in" data-fragment-index="2">
<p><span class="math display">\[
X'_{g_i} = \operatorname{FCN}\left(X_{g_i}\right)
\]</span></p>
<p><span class="math display">\[
X'_G = \left[X'_{g_1}, \cdots, X'_{g_4}\right]
\]</span></p>
</div>
</div>
</div>
<div class="fragment fade-in" data-fragment-index="5">
<div style="font-size: 70%;">
<p><strong>Inter-groups interactions</strong></p>
<div class="callout callout-note callout-style-simple">
<div class="callout-body">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-content">
<p>New representation of each group taking into consideration features from other groups</p>
</div>
</div>
</div>
<p><span class="math display">\[
\begin{align}
\color{query}q_i &amp;= \color{query}X'_{g_i} \cdot W_q^h \\
\color{key}k_i &amp;= \color{key}X'_{g_i} \cdot W_k^h \\
\color{value}v_i &amp;=\color{value} X'_{g_i} \cdot W_v^h
\end{align}
\]</span></p>
<p><span class="math display">\[
\begin{align}
Z &amp;= \operatorname{MultiHeadAttention}\left(X'_G\right) \\
&amp;= \operatorname{concat}\left(\left[h_1, \cdots, h_H \right]\right) \\
h_i &amp;= \operatorname{Attention}\left({\color{query}Q},{\color{key}K},{\color{value}V}\right)
\end{align}
\]</span></p>
</div>
</div>
</div>
</div></div>
<p>:::</p>
<p>:::</p>
<p>::::</p>
</section>
<section id="results" class="slide level2">
<h2>Results</h2>

<img data-src="AttOmicsLimitedTraining.svg" class="r-stretch quarto-figure quarto-figure-center"></section>
<section id="cancer-signature" class="slide level2 smaller">
<h2>Cancer Signature</h2>
<div class="columns">
<div class="column" style="width:70%;">
<p><img data-src="AttOmicsCancerSignatures.png"></p>
</div><div class="column" style="width:30%;">
<ul>
<li>Cancer heatmaps are a mean of patient heatmaps with the same cancer</li>
<li>Each square of the heatmap represents the attention weight between two different group learnt by the model</li>
</ul>
<div class="fragment fade-up" data-fragment-index="1">
<div class="fragment highlight-red" data-fragment-index="1">
<p><strong>Across cancers different interactions are learnt</strong></p>
</div>
</div>
</div></div>
</section>
<section id="cesc-cancer" class="slide level2 smaller">
<h2>CESC cancer</h2>
<div class="columns">
<div class="column" style="width:60%;">
<p><img data-src="AttOmicsCESCPathways.svg" style="width:90.0%"></p>
</div><div class="column" style="width:40%;">
<p><strong>Identified pathways:</strong></p>
<ul>
<li>IL6 JAK STAT3 signaling</li>
<li>Hedgehog signaling</li>
<li>WNT singaling</li>
</ul>
<p><strong>Identified interactions:</strong></p>
<ul>
<li>WNT and Hedgehog cross-talk involved in chemo-resistant cervical cancer</li>
</ul>
</div></div>
</section>
<section id="attomics-conclusions" class="slide level2">
<h2>AttOmics Conclusions</h2>
<ul>
<li>Proposed a grouping mechanism to scale the self-attention to omics data</li>
<li>AttOmics has less parameters and achieves similar or better performances</li>
<li>AttOmics has better performances with a limited training set</li>
<li>Self-attention: capture patient specific feature interactions</li>
</ul>
<div style="font-size=90%;">
<div class="callout callout-important callout-style-simple">
<div class="callout-body">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-content">
<p>Omics were analyzed individually but a phenotype results from their interaction</p>
</div>
</div>
</div>
<div class="callout callout-tip callout-style-simple">
<div class="callout-body">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-content">
<p>Combine the different omics in a single model.</p>
</div>
</div>
</div>
</div>
</section></section>
<section>
<section id="multi-omics-1" class="title-slide slide level1 center">
<h1>Multi omics</h1>

</section>
<section id="multimodal-ai" class="slide level2">
<h2>Multimodal AI</h2>

<img data-src="SOTA_MultiOmics.svg" class="r-stretch"></section>
<section id="attention-as-an-integration-strategy" class="slide level2 smaller">
<h2>Attention as an integration strategy</h2>
<div class="callout callout-note callout-style-simple">
<div class="callout-body">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-content">
<p>Attention mechansim can capture interaction between two vectors</p>
</div>
</div>
</div>
<div class="columns">
<div class="column" style="width:40%;">
<h3 id="early-fusion">Early Fusion</h3>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="AttentionEF.svg" class="quarto-figure quarto-figure-center" style="width:48.0%"></p>
</figure>
</div>
<div class="callout callout-important callout-style-simple">
<div class="callout-body">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-content">
<ul>
<li>High dimensionnality</li>
<li>Attention complexity: <span class="math inline">\(\mathcal{O}(n^2)\)</span></li>
</ul>
</div>
</div>
</div>
</div><div class="column" style="width:60%;">
<h3 id="intermediate-fusion">Intermediate Fusion</h3>
<div class="columns">
<div id="vcenter" class="column" style="width:45%;">
<div>

</div>
<div class="quarto-layout-panel" style="font-size: 80%;" data-layout="[[-1], [-1], [1], [-1]]">
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 100.0%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 100.0%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: flex-start;">
<p><span class="math display">\[\begin{align}
Z_{\beta \rightarrow \alpha} &amp;=  \operatorname{CrossAtt}_{\beta \rightarrow \alpha}\left(X_{\alpha}, X_{\beta} \right) \\
&amp;= \operatorname{Attention}\left(Q_{\alpha},K_{\beta},V_{\beta} \right)
\end{align}\]</span></p>
</div>
</div>
</div>
</div><div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="AttentionIF.svg" class="quarto-figure quarto-figure-center" style="width:80.0%"></p>
</figure>
</div>
</div></div>
<div class="callout callout-important callout-style-simple">
<div class="callout-body">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-content">
<p>Consider all modality pairs: <span class="math inline">\(n(n-1)\)</span> pairs to consider</p>
</div>
</div>
</div>
<div class="callout callout-tip callout-style-simple">
<div class="callout-body">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-content">
<p>Only consider pairs known to interact</p>
</div>
</div>
</div>
</div></div>
</section>
<section id="crossattomics" class="slide level2">
<h2>CrossAttOmics</h2>

<img data-src="CrossAttOmics.svg" class="r-stretch"></section>
<section id="omics-combination" class="slide level2">
<h2>Omics combination</h2>

<img data-src="tcga_perf_omics_comb.svg" class="r-stretch quarto-figure quarto-figure-center"><div class="fragment fade-up">

</div>
</section>
<section id="identifying-important-interactions" class="slide level2">
<h2>Identifying important interactions</h2>
<div class="columns">
<div class="column" style="width:40%;">
<p><strong>LRP</strong></p>
<div data-animate="" data-src="LRP.svg"><!-- setup:
  - element: "#LRP"
    modifier: attr
    parameters:
      - class: fragment
        data-fragment-index: "0" --></div>
<p><img data-src="CrossAttOmicsDetailsLRP.svg" style="width:90.0%"></p>
</div><div class="column" style="width:60%;">
<p><img data-src="CrossAttOmicsLRP.svg" style="width:100.0%"></p>
</div></div>
</section>
<section id="robustness-to-missing-modalities" class="slide level2">
<h2>Robustness to missing modalities</h2>
<div class="columns">
<div class="column" style="width:33%;">
<div>

</div>
<div class="quarto-layout-panel" data-layout="[[1], [1, 1]]">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<p><img data-src="missing_modality_pattern.svg"></p>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<p><img data-src="missing_modality_latent.svg"></p>
</div>
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<p><img data-src="missing_modality_reconstruction.svg"></p>
</div>
</div>
</div>
</div><div class="column" style="width:67%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="CrossAttOmicsModalityDropout.svg" class="quarto-figure quarto-figure-center" style="width:90.0%"></p>
</figure>
</div>
</div></div>
</section>
<section id="crossattomicsgate-a-data-driven-approach" class="slide level2">
<h2>CrossAttOmicsGate: a data-driven approach</h2>
<img data-src="CrossAttOmicsGate.svg" class="r-stretch">
</section>
<section id="crossattomicsgate-results" class="slide level2 smaller">
<h2>CrossAttOmicsGate results</h2>
<table class="caption-top">
<colgroup>
<col style="width: 12%">
<col style="width: 21%">
<col style="width: 21%">
<col style="width: 23%">
<col style="width: 21%">
</colgroup>
<thead>
<tr class="header">
<th>Model</th>
<th style="text-align: right;">Accuracy</th>
<th style="text-align: right;">Precision</th>
<th style="text-align: right;">Recall</th>
<th style="text-align: right;">F1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>No Gate</td>
<td style="text-align: right;">0.980 ± 0.001</td>
<td style="text-align: right;">0.982 ± 0.002</td>
<td style="text-align: right;">0.979 ± 0.002</td>
<td style="text-align: right;">0.980 ± 0.002</td>
</tr>
<tr class="even">
<td>Gate</td>
<td style="text-align: right;">0.987 ± 0.001</td>
<td style="text-align: right;">0.989 ± 0.001</td>
<td style="text-align: right;">0.987 ± 0.001</td>
<td style="text-align: right;">0.987 ± 0.001</td>
</tr>
</tbody>
</table>

<img data-src="CrossAttOmicsGateAlphaHeatmap.svg" class="quarto-figure quarto-figure-center r-stretch"></section>
<section id="multi-omics-conclusions" class="slide level2">
<h2>Multi-omics conclusions</h2>
<div>
<ul>
<li class="fragment">Harness cross-attention to build a multimodal representation that explicitly consider interactions between modalities</li>
<li class="fragment">Using only two or three non-protein omics combination, achieve similar accuracy to what is obtained by training only on proteins</li>
<li class="fragment">Modality dropout is an effective strategy to increase robustness to missing modalities</li>
<li class="fragment">Data-driven strategy to identify important interactions</li>
</ul>
</div>
<div class="callout callout-tip callout-style-simple">
<div class="callout-body">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-content">
<p>Patient can be diagnosed efficiently but what are the disease drivers ? How to treat the patient ?</p>
</div>
</div>
</div>
</section></section>
<section>
<section id="interpretability-1" class="title-slide slide level1 center">
<h1>Interpretability</h1>

</section>
<section id="counterfactuals-properties" class="slide level2">
<h2>Counterfactuals properties</h2>

<img data-src="counterfactualsProperties.svg" class="r-stretch quarto-figure quarto-figure-center"><div>
<ul>
<li class="fragment">Data manifold closeness = respect the original data distribution</li>
<li class="fragment">GANs captures the data distribution (Goodfellow et al., 2014)</li>
</ul>
</div>
</section>
<section id="generating-counterfactuals-with-a-gan" class="slide level2">
<h2>Generating counterfactuals with a GAN</h2>
<img data-src="ConterfactualGAN.svg" class="r-stretch">
<!-- * $\mathcal{L}_{\text{WGAN}} =$ -->
<!-- * $\mathcal{L}_{\operatorname{CL}} = \operatorname{CE}\left(\operatorname{CL}\left(x^{\text{CF}}\right), y^{\text{CF}}\right)$ -->
</section>
<section id="adversarial-training-of-operatornamecl" class="slide level2">
<h2>Adversarial training of <span class="math inline">\(\operatorname{Cl}\)</span></h2>
<div>
<ul>
<li class="fragment">An <strong>adversarial example</strong>, <span class="math inline">\(x^{\star} = x + \delta\)</span>, is an instance <span class="math inline">\(x\)</span> with the addition of a small perturbation <span class="math inline">\(\delta\)</span> that is incorrectly predicted by a model</li>
<li class="fragment">They are obtained by solving <span class="math inline">\(\max_{\delta \in \Delta} \mathcal{L}\left(\operatorname{Cl}\left(x + \delta\right), y\right)\)</span> with gradient ascent. (ex: projected gradient ascent)</li>
<li class="fragment">Adversarial training: <span class="math display">\[
\min_{\operatorname{Cl}} \mathbb{E}_{(x,y)}\left[\max_{\delta^{\star} \in \Delta} \mathcal{L}_{\text{CE}}\left(\operatorname{Cl}\left(x+\delta^{\star}\right), y \right)  \right]
\]</span></li>
</ul>
</div>
</section>
<section id="results-1" class="slide level2">
<h2>Results</h2>
<div class="columns">
<div class="column" style="width:70%;">
<div style="font-size:80%;">
<ul>
<li>Counterfactual Accuracy (<span class="math inline">\(y^{\text{CF}}\)</span> ✅︎ and <span class="math inline">\(y_{\text{Tissue}}\)</span> ✅︎): <span class="math inline">\(\mathcal{A}_{\text{CF}} = 0.96\)</span></li>
<li>Distance to the original point: <span class="math inline">\(L_1\)</span>, <span class="math inline">\(L_2\)</span>, <span class="math inline">\(L_{\infty}\)</span></li>
<li>Sparsity: <span class="math inline">\(L_0 = \sum_i \left|x_i \right|^{0}\)</span> with <span class="math inline">\(\left|0\right|^0 = 0\)</span></li>
<li>kNN Accuracy: <span class="math inline">\(\mathcal{A}_{\text{kNN}} = \frac{1}{k} = \sum_{i=1}^{k} \mathbb{1}\left(y^{\text{CF}}, y_{i}^{\text{kNN}}\right)\)</span></li>
</ul>
</div>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="kNN_Accuracy.svg" class="quarto-figure quarto-figure-center" style="width:50.0%"></p>
</figure>
</div>
</div><div class="column" style="width:30%;">
<div style="font-size:70%;">
<table class="caption-top">
<tbody>
<tr class="odd">
<td><span class="math inline">\(L_1\)</span></td>
<td>2440</td>
</tr>
<tr class="even">
<td><span class="math inline">\(L_2\)</span></td>
<td>30</td>
</tr>
<tr class="odd">
<td><span class="math inline">\(L_{\infty}\)</span></td>
<td>1</td>
</tr>
<tr class="even">
<td><span class="math inline">\(\mathcal{A}_{\text{kNN}}\)</span></td>
<td>0</td>
</tr>
<tr class="odd">
<td><span class="math inline">\(\mathcal{A}_{\text{Oracle}}\)</span></td>
<td>0.94</td>
</tr>
</tbody>
</table>
</div>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="InterClassDistances.svg" class="quarto-figure quarto-figure-center"></p>
</figure>
</div>
</div></div>
</section>
<section id="results-2" class="slide level2">
<h2>Results</h2>

<img data-src="BRCA_CF_UMAP.svg" class="quarto-figure quarto-figure-center r-stretch"></section>
<section id="results-3" class="slide level2">
<h2>Results</h2>

<img data-src="BRCA_CF.svg" class="r-strech quarto-figure quarto-figure-center r-stretch"><div style="font-size:40%;">
<p>GDA: gene-disease association from DisGenet / COSMIC: Catalogue of somatic mutations in cancer</p>
</div>
</section></section>
<section>
<section id="conclusions-and-perspectives-1" class="title-slide slide level1 center">
<h1>Conclusions and Perspectives</h1>

</section>
<section id="conclusions" class="slide level2">
<h2>Conclusions</h2>
<div style="font-size:90%;">
<ul>
<li class="fragment"><strong>AttOmics</strong>: Applied self-attention mechanism to omics profile tp capture patient-specific interactions. Self-attention was applied to groups of features which allowed the addition of knowledge in the groups.</li>
<li class="fragment"><strong>CrossAttOmics</strong>: Integrate multi-omics data based on the known regulatory interactions between modalities and the cross-attention.</li>
<li class="fragment"><strong>CrossAttOmicsGate</strong>: Let the network score each interaction with a gating mechanism</li>
<li class="fragment"><strong>Counterfactual Generation</strong>: Find the perturbation on the molecular profile that will change the prediction from a disease state to a health</li>
</ul>
</div>
</section>
<section id="perspectives-scalar-attention" class="slide level2">
<h2>Perspectives: Scalar Attention</h2>
<div class="columns">
<div class="column" style="width:65%;">
<ul>
<li>In AttOmics, groups are an architectural constraints</li>
<li>Removing groups: directly consider feature interactions</li>
</ul>
<div class="callout callout-important callout-style-simple">
<div class="callout-body">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-content">
<p>How to compute attention between scalar values ?</p>
</div>
</div>
</div>
<p><span class="math display">\[
A_{ij} = \operatorname{softmin}\left(\left|Q_{i} - K_{j} \right| \right)
\]</span></p>
<div class="callout callout-tip callout-style-simple">
<div class="callout-body">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-content">
<p>Efficient implementation with Triton</p>
</div>
</div>
</div>
</div><div class="column" style="width:35%;">
<div class="quarto-figure quarto-figure-right">
<figure>
<p><img data-src="ScalarAttentionMemory.svg" class="quarto-figure quarto-figure-right" style="width:100.0%"></p>
</figure>
</div>
</div></div>
</section>
<section id="perspectives-knowledge" class="slide level2">
<h2>Perspectives: Knowledge</h2>
<ul>
<li>Biolgically-informed architectures are common when applying deep learning to biological problems as they provide a strong inductive bias.</li>
</ul>
<div class="callout callout-important callout-style-simple">
<div class="callout-body">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-content">
<p>Knowledge is incomplete or may contains errors. How to handle this ?</p>
</div>
</div>
</div>
<div class="callout callout-important callout-style-simple">
<div class="callout-body">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-content">
<p>What to do about unnannotated features ?</p>
</div>
</div>
</div>
</section>
<section id="section" class="slide level2 center">
<h2></h2>
<div class="r-fit-text">
<p>Thank you</p>
</div>
</section></section>
<section id="appendix" class="title-slide slide level1 center">
<h1>Appendix</h1>

</section>
    </div>
  <div class="quarto-auto-generated-content" style="display: none;">
<div class="footer footer-default">

</div>
</div></div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="slides_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="slides_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="slides_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="slides_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="slides_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="slides_files/libs/revealjs/plugin/reveal-pointer/pointer.js"></script>
  <script src="slides_files/libs/revealjs/plugin/reveal-animate/plugin.js"></script>
  <script src="slides_files/libs/revealjs/plugin/reveal-animate/svg.min.js"></script>
  <script src="slides_files/libs/revealjs/plugin/reveal-animate/js-yaml.min.js"></script>
  <script src="slides_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="slides_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="slides_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="slides_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="slides_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'jumpToSlide': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleScrollView(event)\"><kbd>r</kbd> Scroll View Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"5\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'pointer': {"key":"q","color":"red","pointerSize":16,"alwaysVisible":false},
'animate': {"autoplay":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1200,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 2.5e-2,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, RevealPointer, RevealAnimate, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const isCodeAnnotation = (el) => {
        for (const clz of el.classList) {
          if (clz.startsWith('code-annotation-')) {                     
            return true;
          }
        }
        return false;
      }
      const onCopySuccess = function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      }
      const getTextToCopy = function(trigger) {
          const codeEl = trigger.previousElementSibling.cloneNode(true);
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
      }
      const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
        text: getTextToCopy
      });
      clipboard.on('success', onCopySuccess);
      if (window.document.getElementById('quarto-embedded-source-code-modal')) {
        // For code content inside modals, clipBoardJS needs to be initialized with a container option
        // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
        const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
          text: getTextToCopy,
          container: window.document.getElementById('quarto-embedded-source-code-modal')
        });
        clipboardModal.on('success', onCopySuccess);
      }
        var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
        var mailtoRegex = new RegExp(/^mailto:/);
          var filterRegex = new RegExp('/' + window.location.host + '/');
        var isInternal = (href) => {
            return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
        }
        // Inspect non-navigation links and adorn them if external
     	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
        for (var i=0; i<links.length; i++) {
          const link = links[i];
          if (!isInternal(link.href)) {
            // undo the damage that might have been done by quarto-nav.js in the case of
            // links that we want to consider external
            if (link.dataset.originalHref !== undefined) {
              link.href = link.dataset.originalHref;
            }
          }
        }
      function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
        const config = {
          allowHTML: true,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start',
        };
        if (contentFn) {
          config.content = contentFn;
        }
        if (onTriggerFn) {
          config.onTrigger = onTriggerFn;
        }
        if (onUntriggerFn) {
          config.onUntrigger = onUntriggerFn;
        }
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note) {
            return note.innerHTML;
          } else {
            return "";
          }
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    <script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox"});
    (function() {
      let previousOnload = window.onload;
      window.onload = () => {
        if (previousOnload) {
          previousOnload();
        }
        lightboxQuarto.on('slide_before_load', (data) => {
          const { slideIndex, slideNode, slideConfig, player, trigger } = data;
          const href = trigger.getAttribute('href');
          if (href !== null) {
            const imgEl = window.document.querySelector(`a[href="${href}"] img`);
            if (imgEl !== null) {
              const srcAttr = imgEl.getAttribute("src");
              if (srcAttr && srcAttr.startsWith("data:")) {
                slideConfig.href = srcAttr;
              }
            }
          } 
        });
      
        lightboxQuarto.on('slide_after_load', (data) => {
          const { slideIndex, slideNode, slideConfig, player, trigger } = data;
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(slideNode);
          }
        });
      
      };
      
    })();
              </script>
    

</body></html>