\documentclass[../main.tex]{subfiles}
%\usepackage{xr}
\usepackage{silence}
\WarningFilter{glossaries}{No \printglossary or \printglossaries found}

\begin{document}

\ifSubfilesClassLoaded{%
	\graphicspath{{figures/1-Introduction/}}%
	\setcounter{chapter}{0}%
	% \externaldocument{chapters/2-Background}%
	% \externaldocument{chapters/3-SOTA}%
	% \externaldocument{chapters/4-AttOmics}%
	% \externaldocument{chapters/5-CrossAttOmics}%
	% \externaldocument{chapters/6-Interpretability}%
	% \externaldocument{chapters/7-Conclusions}%
}{
	\graphicspath{{../figures/1-Introduction/}}%
}

\chapter{Introduction}
\minitoc

\section{Personalized medicine}
 Precision or personalized medicine represents a transformative approach to healthcare, marking a shift from the one-size-fits-all model to one that tailors treatments and medical decisions to individual patients.
 This vision of medicine is not necessarily new, the father of medicine already considered that individual biological factors are responsible for one's pathology: \blockquote[Hippocrates][.]{It is more important to know what sort of person has a disease than to know what sort of disease a person has}.
 This shift has been possible with the development of new data acquisition methods, enabling large patient data collection.
 The collected data includes, but is not limited to:
 \begin{itemize}[nosep]
	 \item \glsxtrfull{ehr},
	 \item imaging data (MRI, scanners, echography, histopathology, etc.),
	 \item smartwatches,
	 \item molecular profiling, also known as omics data.
 \end{itemize}
 The development of high-throughput sequencing methods and their continuous decrease in cost have increased the availability of omics data.
 These omics data provide new insights into the molecular mechanisms involved in pathological conditions.
 Artificial intelligence, and in particular machine learning, is well suited to study such large corpus of data to detect patterns and predicting potential outcomes.
 The application of machine learning on omics data has been pivotal in the development of precision medicine.
 Precision medicine focuses on improving patient health care pathway.
 This improvement is achieved by:
 \begin{itemize}[nosep]
	 \item diagnosing early to provide the best care,
	 \item estimating the patient prognosis, and anticipating disease development,
	 \item predicting the best treatment adequation to a patient.
 \end{itemize}
 The application of machine learning algorithm in precision medicine also allow identifying new biomarkers and improve current knowledge.

 In this thesis, we focused on omics data from cancer patients.
 Cancer is one of the leading causes of death worldwide, with a significant burden on healthcare systems, and its incidence is expected to grow in the following years.
 The complexity and heterogeneity of cancer often limit the effectiveness of conventional therapeutic approaches.
 Furthermore, early diagnosis has been proven important in improving cancer patient survival rates.
 These properties make cancer a good disease to be studied through the prism of precision medicine.

 Omics data encompass data generated by high-throughput sequencing methods to quantify pool of biological molecules.
 This data includes:
 \begin{itemize}[nosep]
	 \item genomics, the study of an organism's entire DNA;
	 \item transcriptomics, the study of the study of all RNA molecules;
	 \item epigenomics, the study of all epigenetic marks;
	 \item proteomics, the study of the entire protein set.
 \end{itemize}
 The DNA molecule, located in the cell nucleus, holds the genetic information of an organism. 
 This molecule is transcribed into various RNA products.
 After a maturation step, a pool of this RNAs becomes \glspl{mrna_mol}. 
 \Glspl{mrna_mol} are exported into the cell cytoplasm where ribosomes will start the translation of the \gls{mrna_mol} into a protein. 
 RNAs producing proteins correspond to coding sequences, or genes,  in the DNA sequence, the non-coding sequences are still transcribed and are involved in the regulation of gene expression. 
 Quantifying all molecular components involved in the correct functioning of a cell generates high-dimensional data. 
 Indeed, there are more than 20 000 known coding genes, more than 30 000 known non-coding genes, each nucleotide position in the DNA sequence can vary between individuals and there are three billions nucleotides in the human genome. 

 The scale of the data highlights the need of computational methods to extract relevant information. 
 Machine learning methods have been successfully applied study omics data and predict phenotypes. 
 However, the high dimensionality of omics data is not compatible with current machine learning models without a prior feature selection. 
 The selection procedure is decoupled from the prediction task and only selected features are used for downstream predictions.
 Using only selected features limits the model's capacity to extract hidden information from the omitted features. 
 
 The last decade has seen a major development of deep learning algorithm which became a standard in many domains such as computer vision or natural language processing.
 Deep learning can extract and exploit the complete information from all features and their interactions. 
 \citeauthor{Hanczar2022} has shown that with the increasing availability of omics data, deep learning architectures are becoming competitive with the state-of-the-art classical machine learning techniques applied to omics data.
 Various deep learning architectures were successfully applied to omics data.
 The application of deep learning to omics data while successful, still face some challenges. 
 The high-dimensionality of omics data lead to very large models, \ie{}with many parameters to estimate, making them susceptible to overfitting.
 With classical deep learning models, the weights used to combine features to construct an abstracted representation are learned during the training phase.
 During the inference phase, they are fixed and assumed to be identical for all patients. 
 However, for a truly personalized medicine features interactions need to be specific for each patient as cellular functions are governed by the combined action of multiple molecular entities which are specific to a patient.

 Focusing on individual omics might not be sufficient to capture the full complexity of biological processes. 
 Indeed, the presence of a mutation in the genome might not correspond to a disease state. 
 Due to the degeneracy of the genetic code, a single mutation can have no effect on the final protein. 
 Moreover, gene expression is highly regulated mechanism. 
 The presence of an \gls{mrna_mol} in a cell does not lead to normal level of protein synthesis. 
 The \gls{mrna_mol} could be targeted by an \gls{mirna}, leading to its degradation, or its half-life could be reduced limiting its capacity to produce proteins. 
 This shows the importance to integrate multi-omics data to improve the predictions made by models.

 Need of Interpretability, IA act ... regulatory but also important for adoption patient/ physician trust the system

\section{Thesis context}
 This Ph.D. is a collaboration between the IBISC laboratory\footnote{\url{https://www.ibisc.univ-evry.fr/}} and Sanofi\footnote{\url{https://www.sanofi.com/}} under the \gls{cifre} program.

 Goal for the lab
 Goal for Sanofi

\section{Thesis objectives}
 \begin{description}[
		 style=multiline,
		 leftmargin=!,
		 labelwidth=\eqboxwidth{listlabel@\EnumitemId}, format=\mylabelformat
	 ]
	 \item[First objective] Classical deep learning models
	 \item[Second objective]
	 \item[Third objective]
 \end{description}



 \clearpage
\section{Thesis overview}
 \begin{center}
	 \ifSubfilesClassLoaded{%
		 \input{figures/1-Introduction/thesis_overview.tex}%
	 }{
		 \input{../figures/1-Introduction/thesis_overview.tex}%
	 }
 \end{center}

 %https://arxiv.org/abs/2206.00520
 %https://www.nature.com/articles/s41568-020-00327-9 % Designing deep learning studies in cancer diagnostics
\end{document}
